### ISSUE TITLE
Bug: Problem with dependencies

### ISSUE THREAD

**Issue by peterteszary** [roots/sage#2949]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Description

The main error message:

composer update
Composer is operating significantly slower than normal because you do not have the PHP curl extension enabled.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires squizlabs/php_codesniffer ^3.6.0 -> satisfiable by squizlabs/php_codesniffer[3.6.0, 3.6.1, 3.6.2, 3.x-dev].
    - squizlabs/php_codesniffer[3.6.0, ..., 3.x-dev] require ext-xmlwriter * -> it is missing from your system. Install or enable PHP's xmlwriter extension.

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.3/cli/php.ini
    - /etc/php/7.3/cli/conf.d/10-opcache.ini
    - /etc/php/7.3/cli/conf.d/10-pdo.ini
    - /etc/php/7.3/cli/conf.d/20-calendar.ini
    - /etc/php/7.3/cli/conf.d/20-ctype.ini
    - /etc/php/7.3/cli/conf.d/20-exif.ini
    - /etc/php/7.3/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.3/cli/conf.d/20-ftp.ini
    - /etc/php/7.3/cli/conf.d/20-gettext.ini
    - /etc/php/7.3/cli/conf.d/20-iconv.ini
    - /etc/php/7.3/cli/conf.d/20-json.ini
    - /etc/php/7.3/cli/conf.d/20-phar.ini
    - /etc/php/7.3/cli/conf.d/20-posix.ini
    - /etc/php/7.3/cli/conf.d/20-readline.ini
    - /etc/php/7.3/cli/conf.d/20-shmop.ini
    - /etc/php/7.3/cli/conf.d/20-sockets.ini
    - /etc/php/7.3/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.3/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.3/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.3/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-xmlwriter` to temporarily ignore these required extensions.

### Steps To Reproduce

USING POP OS and LOCAL by Flywheel:

With this tutorial:
https://www.youtube.com/watch?v=Mfm-qUjbrA4&ab_channel=DenverCoder

Local by flywheel
https://localwp.com/releases/6.1.8/

The environment in LOCAL:
PHP 7.3.5

WordPress version: 
5.8.3

-CLEAR EVERYTHING BEFORE STARTING-

Remove php:
https://askubuntu.com/questions/59886/how-to-completely-remove-php

Install php 7.3
https://computingforgeeks.com/how-to-install-php-ubuntu-debian/

Install composer: https://www.digitalocean.com/community/tutorials/how-to-install-and-use-composer-on-ubuntu-20-04

Installing node.js:
https://ashokma.com/2019/02/18/install-nodejs-and-npm-pop-os/

Install yarn:
https://classic.yarnpkg.com/en/docs/install#debian-stable

Install PHP_CodeSniffer
https://packagist.org/packages/squizlabs/php_codesniffer

### Expected Behavior

That it is working.

### Actual Behavior

I get the error message added above.

### Versions

Pop!_OS 21.10

**Comment by peterteszary** [comment]
**Started with a clean install again with this:**

https://docs.roots.io/getting-started/ubuntu-linux/#global-dependencies

And after all with this:

- https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent
- https://docs.github.com/en/authentication/connecting-to-github-with-ssh/working-with-ssh-key-passphrases
- Edit Configuration File
- Open your SSH configuration file with the command: sudo gedit /etc/ssh/sshd_config.
- https://tecadmin.net/how-to-install-nvm-on-ubuntu-20-04/
- https://ashokma.com/2019/02/18/install-nodejs-and-npm-pop-os/
- https://classic.yarnpkg.com/lang/en/docs/install/#debian-stable
- https://www.vagrantup.com/downloads
- https://linuxize.com/post/how-to-install-pip-on-ubuntu-20.04/
- https://www.digitalocean.com/community/tutorials/how-to-install-and-use-composer-on-ubuntu-20-04

**I still got error message like this:**

Creating a "roots/sage" project at "./newdawn"
Installing roots/sage (9.0.10)
  - Installing roots/sage (9.0.10): Extracting archive
Created project in /home/peterteszary/Local Sites/newdawn/app/public/wp-content/themes/newdawn
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - illuminate/support is locked to version v5.6.39 and an update of this package was not requested.
    - illuminate/support v5.6.39 requires ext-mbstring * -> it is missing from your system. Install or enable PHP's mbstring extension.
  Problem 2
    - squizlabs/php_codesniffer is locked to version 2.9.2 and an update of this package was not requested.
    - squizlabs/php_codesniffer 2.9.2 requires ext-simplexml * -> it is missing from your system. Install or enable PHP's simplexml extension.
  Problem 3
    - illuminate/support v5.6.39 requires ext-mbstring * -> it is missing from your system. Install or enable PHP's mbstring extension.
    - illuminate/console v5.6.39 requires illuminate/support 5.6.* -> satisfiable by illuminate/support[v5.6.39].
    - illuminate/console is locked to version v5.6.39 and an update of this package was not requested.

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.3/cli/php.ini
    - /etc/php/7.3/cli/conf.d/10-opcache.ini
    - /etc/php/7.3/cli/conf.d/10-pdo.ini
    - /etc/php/7.3/cli/conf.d/20-calendar.ini
    - /etc/php/7.3/cli/conf.d/20-ctype.ini
    - /etc/php/7.3/cli/conf.d/20-exif.ini
    - /etc/php/7.3/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.3/cli/conf.d/20-ftp.ini
    - /etc/php/7.3/cli/conf.d/20-gettext.ini
    - /etc/php/7.3/cli/conf.d/20-iconv.ini
    - /etc/php/7.3/cli/conf.d/20-json.ini
    - /etc/php/7.3/cli/conf.d/20-phar.ini
    - /etc/php/7.3/cli/conf.d/20-posix.ini
    - /etc/php/7.3/cli/conf.d/20-readline.ini
    - /etc/php/7.3/cli/conf.d/20-shmop.ini
    - /etc/php/7.3/cli/conf.d/20-sockets.ini
    - /etc/php/7.3/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.3/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.3/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.3/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-mbstring --ignore-platform-req=ext-simplexml --ignore-platform-req=ext-mbstring` to temporarily ignore these required extensions.

**Comment by teppokoivula** [comment]
Hey @peterteszary. The issues you're seeing are related to your local environment, not _directly_ Sage. You are missing a number of PHP extensions which are required to install the dependencies of Sage.

From your report:

> Install php 7.3
> https://computingforgeeks.com/how-to-install-php-ubuntu-debian/

Please make sure that you follow the steps detailed in "Step 3: Installing PHP 7.3 Extensions". The example they've provided looks like a good starting point, it should get you most of the required extensions.

You can see the missing extensions in the output from Composer, in case you're still missing some.

**Comment by peterteszary** [comment]
I have tried everything but it is just not working.
Any scripts or advice?
Thanks

**Comment by teppokoivula** [comment]
There's a command for installing dependencies in the article linked above — check "Step 3: Installing PHP 7.3 Extensions" and check the example there ("See example below").

Either way this is not a Sage bug, but rather seems to fall under the "personal support request" category, so if you're still having issues, you should probably open a thread for it at https://discourse.roots.io/.

**Comment by peterteszary** [comment]
Hello again,

So, it is not working because of Pop OS. It has some compatibility issues with this OS. I have installed Sage in a docker file and it is kind of working now. I've just started to explore it, but now I could install it at least.
